libs/debugging/include/hpx/debugging/print.hpp

```diff
@@ -225,71 +307,201 @@ namespace hpx { namespace debug {
             tuple_print(os, tup, std::make_index_sequence<sizeof...(Args)>());
         }
 
+#else
+        // C++14 version
+        // helper function to print a tuple of any size
+        template <typename TupleType, std::size_t N>
+        struct tuple_printer
+        {
+            static void print(std::ostream& os, const TupleType& t)
+            {
+                tuple_printer<TupleType, N - 1>::print(os, t);
+                os << " " << std::get<N - 1>(t);
+            }
+        };
+
+        template <typename TupleType>
+        struct tuple_printer<TupleType, 1>
+        {
+            static void print(std::ostream& os, const TupleType& t)
+            {
+                os << std::get<0>(t);
+            }
+        };
+
+        template <class... Args>
+        void tuple_print(std::ostream& os, const std::tuple<Args...>& t)
+        {
+            tuple_printer<decltype(t), sizeof...(Args)>::print(os, t);
+        }
```

Comment:
FWIW, something like the following would avoid the recursive expansion:
```suggestion
        // C++14 version
        // helper function to print a tuple of any size
        namespace detail {
            template <typename Tuple, std::size_t... Is>
            void tuple_print(std::ostream& os, Tuple const& t, 
                 std::index_sequence<Is...>)
            {
                using expander = int[];
                (void) expander{0, (void(os << ' ' << std::get<Is>(t)), 0)...};
            }
       }
        
        template <typename... Args> 
        void tuple_print( 
            std::ostream& os, std::tuple<Args...>const& t) 
        { 
            detail::tuple_print(os, t, std::make_index_sequence<sizeof...(Args)>{});
        }
```
```

Reply:
I'll try this out, thanks